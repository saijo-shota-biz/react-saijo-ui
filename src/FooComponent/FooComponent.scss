.foo {
  font-size: 10rem;
}
