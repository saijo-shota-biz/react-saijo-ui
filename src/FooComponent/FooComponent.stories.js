import React from "react";
import FooComponent from "./FooComponent";

export default {
  title: "FooComponent",
};

export const red = () => <FooComponent color="red" />;

export const blue = () => <FooComponent color="blue" />;
