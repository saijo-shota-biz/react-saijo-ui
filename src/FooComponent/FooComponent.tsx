import React from "react";
import "./FooComponent.scss";

export type FooComponentProp = {
  color: "red" | "blue";
};

const FooComponent: React.FC<FooComponentProp> = ({ color }) => {
  return (
    <div className="foo" style={{ color }}>
      Foo
    </div>
  );
};

export default FooComponent;
