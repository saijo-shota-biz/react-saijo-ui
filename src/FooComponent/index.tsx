import FooComponent from "./FooComponent";

export { FooComponent };

export default FooComponent;
