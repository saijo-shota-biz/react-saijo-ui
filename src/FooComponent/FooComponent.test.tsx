import React from "react";
import { shallow } from "enzyme";

import FooComponent, { FooComponentProp } from "./FooComponent";

describe("Test Component", () => {
  let props: FooComponentProp;

  beforeEach(() => {
    props = {
      color: "red",
    };
  });

  const renderWrapper = () => shallow(<FooComponent color={props.color} />);

  describe("Snapshots", () => {
    it("should match snapshots as primary themed", () => {
      expect(renderWrapper()).toMatchSnapshot();
    });

    it("should match snapshots as secondary themed", () => {
      props.color = "blue";
      expect(renderWrapper()).toMatchSnapshot();
    });
  });
});
