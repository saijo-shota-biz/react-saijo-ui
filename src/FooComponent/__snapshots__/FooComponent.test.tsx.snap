// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Snapshots should match snapshots as primary themed 1`] = `
<div
  className="foo"
  style={
    Object {
      "color": "red",
    }
  }
>
  Foo
</div>
`;

exports[`Test Component Snapshots should match snapshots as secondary themed 1`] = `
<div
  className="foo"
  style={
    Object {
      "color": "blue",
    }
  }
>
  Foo
</div>
`;
